lib/timeline/component/ItemSet.js

```diff
@@ -1662,21 +1685,19 @@ ItemSet.prototype._onGroupClick = function (event) {
 
   if (!group || !group.nestedGroups) return;
 
-  var groupsData = this.groupsData;
-  if (this.groupsData instanceof DataView) {
-    groupsData = this.groupsData.getDataSet()
-  }
+  var groupsData = this.groupsData.getDataSet();
 
-  group.showNested = !group.showNested;
+  var nestingGroup = groupsData.get(group.groupId)
+  nestingGroup.showNested = !nestingGroup.showNested;
 
   var nestedGroups = groupsData.get(group.nestedGroups).map(function(nestedGroup) {
-    if (nestedGroup.visible == undefined) { nestedGroup.visible = true; }
-    nestedGroup.visible = !!group.showNested;
+    nestedGroup.visible = nestingGroup.showNested;
     return nestedGroup;
   });
-  groupsData.update(nestedGroups);
 
-  if (group.showNested) {
+  groupsData.update(nestedGroups.concat(nestingGroup));
```

Comment:
I think it's a bit problematic to not have a sender here. But introducing something like a fake sender like 'nesting' wouldn't be clean either. My problem is that I catch the update event on the groups and do a lot of stuff afterwards. If there's no sender, I assume that the user modified the group, but in this case the user only cliked on the expand/collapse icon and didn't change the groups themselves. Let me illustrate this:

```
that.groups.on('update', function (event, properties, senderId) {
	switch (senderId) {
	case 'api':
		// do nothing / accept the changes
		break;
	case 'initialization':
		break;
	default:
		// item is supplied locally from the user/the library
		let currObj = that.groups.get(group.id);
		// --> update the changes to the API
	}
});
```
With your changes, I may now perform a rather complicated object comparison to see if the changes were only in the visibility of the group, but that's a bit cumbersome. Nevertheless, I have no good idea on how to approach this and I can live with it.